# HG changeset patch
# User Rik <rik@octave.org>
# Date 1522269708 25200
#      Wed Mar 28 13:41:48 2018 -0700
# Branch stable
# Node ID d5d61f1b6de09546cb1da28978d4d8f94b0f9c4b
# Parent  328518881291d7e6b175b1718641a381038c2418
Deprecate java2mat function.

* NEWS: Announce deprecation.

* scripts/deprecated/java2mat.m: New function to issue a warning if used.  Calls
__java2mat__ to implement functionality.

* scripts/deprecated/module.mk: Add java2mat.m to build system.

* ov-java.cc (F__java2mat__): Renamed from java2mat.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -302,16 +302,18 @@ Summary of important user-visible change
     release after 4.4):
 
       Function             | Replacement
       ---------------------|------------------
       chop                 | sprintf for visual results
       desktop              | isguirunning
       tmpnam               | tempname
       toascii              | double
+      java2mat             | __java2mat__
+
 
  ** The following functions were deprecated in Octave 4.0 and have been
     removed from Octave 4.4.
 
       allow_noninteger_range_as_index
       bicubic
       delaunay3
       do_braindead_shortcircuit_evaluation
diff --git a/libinterp/octave-value/ov-java.cc b/libinterp/octave-value/ov-java.cc
--- a/libinterp/octave-value/ov-java.cc
+++ b/libinterp/octave-value/ov-java.cc
@@ -1843,17 +1843,17 @@ java_event_hook (void)
   return 0;
 }
 
 //! Initialize java including the virtual machine (jvm) if necessary.
 //!
 //! Initializes the fields #jvm, #jvm_attached, #jvm_lib, and
 //! #octave_thread_ID.  To ensure that java is initialized, this method is
 //! used as part of octave functions @c javaObject, @c javaMethod,
-//! @c __java_get__, @c __java_set__, and @c java2mat.
+//! @c __java_get__, @c __java_set__, and @c __java2mat__.
 
 static void
 initialize_java (void)
 {
   if (! jvm)
     {
       try
         {
@@ -3175,19 +3175,19 @@ equivalent
 
   octave_unused_parameter (args);
 
   err_disabled_feature ("__java_set__", "Java");
 
 #endif
 }
 
-DEFUN (java2mat, args, ,
+DEFUN (__java2mat__, args, ,
        doc: /* -*- texinfo -*-
-@deftypefn {} {} java2mat (@var{javaobj})
+@deftypefn {} {} __java2mat__ (@var{javaobj})
 Undocumented internal function.
 @end deftypefn */)
 {
 #if defined (HAVE_JAVA)
 
   if (args.length () != 1)
     print_usage ();
 
@@ -3206,17 +3206,17 @@ Undocumented internal function.
     retval = ovl (args(0));
 
   return retval;
 
 #else
 
   octave_unused_parameter (args);
 
-  err_disabled_feature ("java2mat", "Java");
+  err_disabled_feature ("__java2mat__", "Java");
 
 #endif
 }
 
 DEFUN (java_matrix_autoconversion, args, nargout,
        doc: /* -*- texinfo -*-
 @deftypefn  {} {@var{val} =} java_matrix_autoconversion ()
 @deftypefnx {} {@var{old_val} =} java_matrix_autoconversion (@var{new_val})
diff --git a/scripts/deprecated/java2mat.m b/scripts/deprecated/java2mat.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/java2mat.m
@@ -0,0 +1,43 @@
+## Copyright (C) 2018 John W. Eaton
+##
+## This file is part of Octave.
+##
+## Octave is free software: you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <https://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {} {} java2mat (@var{javaobj})
+## @code{java2mat} is deprecated and will be removed in Octave version 4.8.
+##
+## For the next two releases, use @code{__java2mat__} if necessary, and file
+## a bug report explaining your programming use of @code{java2mat} and how it
+## can't be done with other Octave functions.
+## @end deftypefn
+
+function retval = java2mat (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "java2mat is obsolete and will be removed from a future version of Octave");
+  endif
+
+  retval = __java2mat__ (varargin{:});
+
+endfunction
+
+
+## No tests needed for alias.
+%!assert (1)
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -1,16 +1,17 @@
 FCN_FILE_DIRS += scripts/deprecated
 
 %canon_reldir%_FCN_FILES = \
   %reldir%/bitmax.m \
   %reldir%/chop.m \
   %reldir%/comma.m \
   %reldir%/desktop.m \
   %reldir%/isstr.m \
+  %reldir%/java2mat.m \
   %reldir%/mahalanobis.m \
   %reldir%/md5sum.m \
   %reldir%/octave_config_info.m \
   %reldir%/onenormest.m \
   %reldir%/paren.m \
   %reldir%/semicolon.m \
   %reldir%/sleep.m \
   %reldir%/tmpnam.m \
